odoo/api.py

```diff
@@ -34,12 +33,31 @@
 from .tools.translate import _
 from odoo.tools.misc import StackMap
 
-if TYPE_CHECKING:
+import typing
+if typing.TYPE_CHECKING:
+    from collections.abc import Callable
     from odoo.sql_db import Cursor, TestCursor
     from odoo.models import BaseModel
+    from .models import NewId
+
+    try:
+        from typing_extensions import Self  # noqa: F401
+    except ImportError:
+        from typing import Self  # noqa: F401
+    M = typing.TypeVar("BaseModel", bound=BaseModel)
+    IdType = int | NewId
```

Comment:
Probably a bug with mypy, but when you import `IdType` it is not considered valid as a type. The workaround I found is:
```suggestion
    # type written as union because not recognized by mypy when imported
    IdType = typing.Union[int, NewId]  # noqa: UP007
```